service/src/main/java/org/wildfly/service/Dependency.java

```diff
@@ -18,6 +19,28 @@
  */
 public interface Dependency<B extends ServiceBuilder<?>, V> extends Consumer<B>, Supplier<V> {
 
+    @Override
+    default Dependency<B, V> andThen(Consumer<? super B> after) {
+        return new Dependency<>() {
```

Comment:
Should not do a sanity check `Objects.requireNonNull(after);`?

Reply:
Sounds good.